app/models/item_pool.rb

```diff
@@ -0,0 +1,71 @@
+class ItemPool < ApplicationRecord
+  belongs_to :creator, class_name: "User"
+  has_many :reservable_items
+  has_many :date_holds
+
+  validates :name, presence: true
+  validate :unnumbered_has_no_items
+
+  scope :uniquely_numbered, -> { where(uniquely_numbered: true) }
+
+  acts_as_tenant :library
+
+  def max_reservable_quantity
+    if uniquely_numbered
+      reservable_items_count
+    else
+      unnumbered_count
+    end
+  end
+
+  ## Return a Hash of date => reserved quantity for every day in the range provided
+  def reserved_by_date(starts, ends, ignored_reservation_id: nil)
+    totals_by_day = sum_reservations_by_day(starts, ends, ignored_reservation_id:)
+    starts.to_date.upto(ends.to_date).each_with_object({}) do |date, hash|
+      hash[date] = totals_by_day[date] || 0
+    end
+  end
+
+  # Return the maximum number of items that are available for reservation for the entire duration
+  # between starts and ends.
+  def max_available_between(starts, ends, ignored_reservation_id: nil)
+    total_items = max_reservable_quantity
+
+    totals_by_day = sum_reservations_by_day(starts, ends, ignored_reservation_id:)
+
+    if totals_by_day.empty?
+      total_items
+    else
+      total_items - totals_by_day.values.max
+    end
+  end
+
+  private
+
+  # Return a Hash of date => total reservations for the given range.
```

Comment:
There is some duplication in this file still. This method I feel pretty good about, the other ones that wrap it I'm not as sure about. I figure that stuff will get ironed out as we start to build out real UIs for the new flows.